### PR TITLE
feat(landing-page): new 2021 landing page

### PR DESCRIPTION
**Pull request by mattnolting**
New landing page

Docs: 
- https://docs.google.com/document/d/16mrk2gORQUgVp13EbJN5_GbkKDDkRdaPprQaogZY1ww/edit
- https://xd.adobe.com/view/e9237ffa-089b-471a-b697-1882d52e9b78-bc70/screen/204eea27-e82b-4c3c-8c3b-4a6b71b0ead9/specs/

Screen Shot 2021-04-16 at 7 59 13 AM

Screen Shot 2021-04-16 at 7 59 21 AM

Screen Shot 2021-04-16 at 7 59 32 AM

### CODE REVIEW

File: src/routes/Landing.scss [hunk 1]
```diff
@@ -117,9 +119,10 @@ sup.ins-c-rball { @include rem('font-size', 10px); }
     }
 }
 
-// Adjust PF layout to be able to display witouth header
+// Adjust PF layout to be able to display whithout header
```
Comment (epwinchell) [inline src/routes/Landing.scss:122]: without :)

**Review by epwinchell — APPROVED** [review]
@mattnolting Very nice! Tested in FF, Safari and Chrome at all breakpoints.

**Comment by Hyperkid123** [comment]
@mattnolting can you fix up the tests, please? I looked through the code briefly and I was wondering how do we decide what links open a new window/tab? I see some links to redhat.com are external and some are not.

Also, do we have some mocks or a preview or did you make it from scratch?

File: src/components/HeroIcon.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './HeroIcon.scss';
+
+const HeroIcon = ({ src, alt }) => (
+  <div className="ins-c-hero__title-icon">
```
Comment (karelhala) [inline src/components/HeroIcon.js:6]: This is using flex layout right? Might be more reasonable to use the layout
```JSX
import { Flex, FlexItem } from '@patternfly/react-core';
const HeroIcon = (props) => (
  <Flex className="ins-c-hero__title-icon">
    <img {...props} aria-hidden="true" />
  </Flex>
);
```

The only thing that will be still required is the `width`.

File: src/components/marketing/icon__build-deploy.svg [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (karelhala) [inline src/components/marketing/icon__build-deploy.svg:1]: No need to include SVGs in this repository we have https://github.com/RedHatInsights/frontend-assets repository to host these files.

File: src/layout/FooterMarketing.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { Brand, Flex, FlexItem, PageSection } from '@patternfly/react-core';
+import React from 'react';
+import logoInsights from './images/logo__redhat.svg';
+
+import './FooterMarketing.scss';
+
+const FooterTraditional = () => {
+  return (
+    <React.Fragment>
```
Comment (karelhala) [inline src/layout/FooterMarketing.js:9]: Extra `return` and `React.Fragment` No need to use these as you have just one `PageSection`

File: src/layout/FooterMarketing.scss [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    }
+  }
+
+  .ins-c-footer-marketing__logo-link {
+    display: flex;
+  }
+
+  *,
+  a {
+    color: #aaa;
```
Comment (karelhala) [inline src/layout/FooterMarketing.scss:27]: I don't really like custom colors, can you use `--pf-global--palette--black-500` or `--pf-global--palette--black-600` for this one? Or are they just way too off for this place?

File: src/layout/FooterMarketing.scss [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+@import '~@patternfly/patternfly/sass-utilities/_all';
+
+.ins-c-footer-marketing {
+  --pf-c-page--section--m-limit-width--MaxWidth: #{pf-size-prem(1140px)};
+  --pf-c-page__main-section--PaddingTop: var(--pf-global--spacer--md);
+  --pf-c-page__main-section--PaddingRight: var(--pf-global--spacer--md);
+  --pf-c-page__main-section--PaddingBottom: var(--pf-global--spacer--md);
+  --pf-c-page__main-section--PaddingLeft: var(--pf-global--spacer--md);
+
+  background-color: #1a1a1a;
```
Comment (karelhala) [inline src/layout/FooterMarketing.scss:11]: I think the closes PF color to this one is `--pf-global--palette--black-900` can we use it here?

File: src/layout/FooterMarketing.scss [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  a {
+    color: #aaa;
+  }
+
+  a {
+    text-decoration: underline;
+  }
+
+  a:hover {
+    color: #fff;
```
Comment (karelhala) [inline src/layout/FooterMarketing.scss:35]: Same as above, can we use PF var color `--pf-global--palette--black-100`

File: src/layout/Hero.js [hunk 8]
```diff
@@ -1,38 +1,71 @@
...
+        {/* Activate when support is available
+        <Button
+          className="ins-m-marketing"
+          variant="secondary"
+          component="a"
+          isLarge
+          href={heroContent.secondaryCTAurl}
+        >
+          {heroContent.secondaryCTAtext}
+        </Button> */}
```
Comment (karelhala) [inline src/layout/Hero.js:66]: No need to keep commented out code. Github tracks the history for us.

File: src/layout/Hero.scss [hunk 9]
```diff
@@ -2,20 +2,71 @@
...
+
+    &::before {
+        position: absolute;
+        right: 0;
+        top: 0;
+        bottom: 0;
+        width: var(--ins-c-hero--before--Width);
+        content: '';
+        opacity: var(--ins-c-hero--before--Opacity);
+        background-image: #{url('https://raw.githubusercontent.com/RedHatInsights/frontend-assets/0331b39e1a370e5f1c243452330df61f4e1b57bc/src/background-images/bg__hero.svg')};
```
Comment (karelhala) [inline src/layout/Hero.scss:29]: You don't have to use `githubusercontent` as URL, just `https://cloud.redhat.com/apps/frontend-assets/src/background-images/bg__hero.svg` is the URL correct @ryelo ?

File: src/layout/Marketing.scss [hunk 10]
```diff
@@ -1,45 +1,172 @@
...
+
+        // Primary
+        --pf-c-button--m-primary--Color: #fff;
+        --pf-c-button--m-primary--BackgroundColor: #ee0000;
+
+        // Secondaery
+        --pf-c-button--m-secondary--Color: #151515;
+        --pf-c-button--m-secondary--after--BorderColor: #151515;
+        --pf-c-button--m-secondary--hover--after--BorderColor: #151515;
+        --pf-c-button--m-secondary--BackgroundColor: transparent;
```
Comment (karelhala) [inline src/layout/Marketing.scss:27]: Can we use PF colors for these values?

**Review by karelhala — COMMENTED** [review]
Looking good! I have just a few nitpicks about using custom colors instead of PF ones and perhaps moving SVGs to frontend-assets repo and using the full URL for them.

**Comment by ryelo** [comment]
@karelhala not trying to speak for Matt here, but I believe this page is actually using branded Red Hat colors + assets (not Patternfly), so there may be some inconsistencies with the color mapping.

This could also totally be wrong 😄

File: src/layout/Hero.scss (hunk 9 above)
Comment (ryelo) [reply src/layout/Hero.scss:29]: Correct

File: src/consts.js [hunk 11]
```diff
@@ -1,41 +1,287 @@
...
+import iconPadlock from './components/marketing/icon__padlock.svg';
+import iconPrivateCloud from './components/marketing/icon__private-cloud.svg';
+import iconUpdate from './components/marketing/icon__update.svg';
+import imgPlaceholder from './layout/images/img__placeholder.png';
 
-import insightsMarketing from './components/marketing/insightsMarketing.svg';
-import openShiftMarketing from './components/marketing/openShiftMarketing.svg';
-import ansibleMarketing from './components/marketing/ansibleMarketing.svg';
+export const productGridHeader = 'Build your cloud on open source';
+export const productGridLoginText = 'Login in to the console';
```
Comment (ryelo) [inline src/consts.js:24]: `Login in`

**Review by ryelo — CHANGES_REQUESTED** [review]

**Comment by mattnolting** [comment]
> @karelhala not trying to speak for Matt here, but I believe this page is actually using branded Red Hat colors + assets (not Patternfly), so there may be some inconsistencies with the color mapping.
> 
> This could also totally be wrong 😄

That's _pretty much_ correct. I've updated all colors we currently provide w/exception for marketing specific values and added notes for those custom assignments.

**Comment by Hyperkid123** [comment]
@mattnolting I've added a commit that fixes up the tests and build. There is a list of changes

- some of the unAuthed assertions were irrelevant due to the technology data change. I have removed obsolete assertions but kept as much as possible
- One major issue was the different Modal root element. I've mocked the root element to the Landing test file
- Fixed a bunch of linting errors
- Added safe accessor to nested objects which might be undefined

Please check if the visuals did not change in any way (they should not)

**Comment by mattnolting** [comment]
@Hyperkid123 Looks awesome, thank you!!

File: src/layout/GetStarted.js [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+          className="ins-c-section-title ins-m-align-center"
+        >
+          {getStarted.title}
+        </Title>
+      </FlexItem>
+      <FlexItem>
+        <Button
+          className="ins-m-marketing"
+          variant="primary"
+          isLarge
```
Comment (john-dupuy) [inline src/layout/GetStarted.js:38]: Is this the main login button? If so, can we add an ouiaId prop to this one? QE can use it to easily locate the button within the DOM. 

It is enough to just add a prop like
```
ouiaId="primary-login-button"
```

For the top right login button, it'd be nice if we had some ouiaId like
```
ouiaId="top-right-login-button"
```

Comment (ryelo) [reply src/layout/GetStarted.js:38]: @john-dupuy top right one is coming from chrome

Comment (john-dupuy) [reply src/layout/GetStarted.js:38]: oh yeah! let me go add an id to that one

**Comment by ryelo** [comment]
Yeah this LGTM when you can add that OUIA attribute

**Comment by mattnolting** [comment]
@ryelo @john-dupuy Updated

File: src/layout/GetStarted.js (hunk 12 above)
Comment (john-dupuy) [reply src/layout/GetStarted.js:38]: Thanks!

**Review by ryelo — APPROVED** [review]

**Pull request merged**